pytorch_lightning/accelerators/gpu.py

```diff
@@ -48,6 +48,9 @@ def on_train_start(self) -> None:
             torch.cuda.empty_cache()
 
     def teardown(self) -> None:
+        # Note: this would be problematic for large model (which could not fit in one GPU)
+        # as FSDP module.to(device) would first summon all parameters
+        # (TODO: need to figure out solution)
         self.lightning_module.cpu()
```

Comment:
@awaelchli @justusschock this is pointing out that teardown will be training type specific, not necessarily just device specific. if we're rethinking this in https://github.com/PyTorchLightning/pytorch-lightning/issues/7324 i think combining training type and accelerator, or parameterizing training type to accept what devices are being used, will make this easier to work through